pkg/skaffold/initializer/deploy/init.go

```diff
@@ -25,12 +25,12 @@ type Error string
 
 func (e Error) Error() string { return string(e) }
 
-const NoManifest = Error("one or more Kubernetes manifests are required to run skaffold")
+const NoManifest = Error("one or more valid Kubernetes manifests are required to run skaffold")
 
 // Initializer detects a deployment type and is able to extract image names from it
 type Initializer interface {
 	// deployConfig generates Deploy Config for skaffold configuration.
-	DeployConfig() latest.DeployConfig
+	DeployConfig() (latest.DeployConfig, []latest.Profile)
```

Comment:
nit: shd we rename `DeployConfig` to `DeployConfigWithProfiles` since profile is not part of deploy config.
Or else, we could define a new method `Profiles`.

Upto you.

Reply:
I thought about this, but in the end decided it was ok since the contract is defined in the interface and is documented. `DeployConfigWithProfiles` is technically correct but seemed too verbose to me, especially because it's getting called next to `BuildConfig`. I'd vote to leave it alone unless you feel strongly.

Reply:
The alternative is a separate `Profiles()`.

Reply:
this would also need a big refactor since the profiles get generated basically from excess deploy configuration. i'm not really sure the best way to handle it 🤷